### ISSUE TITLE
tiny_yolo_voc can be trained normally, but yolo_voc's  precision & recall are  0.000000 all the time while training.

### ISSUE THREAD

**Issue by xiaozhi2015** [marvis/pytorch-yolo2#64]

**Comment by xiaozhi2015** [comment]
Anyone came across this problem?

**Comment by denck007** [comment]
I just started using this piece of code and am having the same issue on tiny-yolo-voc and yolo-voc where the precision, recall, and fscore are all 0. What I found was that the test code has some bugs:

- The test_loader does not return any labels. The train flag is set to false, so it tries to find the label file in a different way than the train mode does, and it fails if you do not have your folders set up exactly like the creator (it uses string replace on the path). It is trying to preallocate the tensor with zeros, and then never finds the correct values so it just returns zeros.
- The loops in the test function all have the same indexer, i, which is going to cause some problems. 

To fix this I have:
- Set the train flag in the test_loader dataset to True, it should work properly. [ IE: Change this line to True](https://github.com/marvis/pytorch-yolo2/blob/6c7c1561b42804f4d50d34e0df220c913711f064/train.py#L93)
- Set the loops in the test function to use independent indexers.

**Comment by xiaozhi2015** [comment]
@denck007   But when I train tiny_yolo_voc, the precision & recall are normal.

**Comment by wwxJream** [comment]
@xiaozhi2015 I find that when I train `yolo_voc`, the precision & recall are normal. But when I try  `tiny_yolo_voc`, the precision & recall are 0. By the way, what's mAP you get when train `tiny_yolo_voc`?

**Comment by xiaozhi2015** [comment]
@wwxJream 46%, lower than official website. I think the reason is batch_size. By the way, which place have you changed this scorce code when you trained a yolo_voc model?

**Comment by wwxJream** [comment]
@xiaozhi2015 No, I change nothing. I find that you need more patience to train. In the first epoch, precision & recall is 100+, and later they become 0. After 30 epochs, they will become 50+. So, be patient !!

**Comment by wwxJream** [comment]
@xiaozhi2015 I am studying yolo, too. We can talk about it. Here is my QQ:513764295

**Comment by xiaozhi2015** [comment]
3q

**Comment by xiaozhi2015** [comment]
3q
